Sources/Mustache/Sequence.swift

```diff
@@ -13,16 +13,31 @@
 //===----------------------------------------------------------------------===//
 
 /// Protocol for objects that can be rendered as a sequence in Mustache
-protocol MustacheSequence {
+///
+/// You're not supposed to manually conform to this protocol.
+/// Instead, conform your type to `Sequence`, and then declare conformance to `MustacheSequence` too.
+/// This way you'll get the automatically-synthesized conformance of `Sequence` to `MustacheSequence`.
+///
+/// ```swift
+/// import Mustache
+///
+/// struct MyCustomMustacheSequence: Sequence {
+/// /// Provide necessary declarations so the type conforms to `Sequence`
+/// }
+///
+/// /// Receive the automatically-synthesized conformance to `MustacheSequence` for free.
+/// extension MyCustomMustacheSequence: MustacheSequence {}
+/// ```
+public protocol MustacheSequence: Sequence {
     /// Render section using template
     func renderSection(with template: MustacheTemplate, context: MustacheContext) -> String
     /// Render inverted section using template
     func renderInvertedSection(with template: MustacheTemplate, context: MustacheContext) -> String
 }
```

Comment:
If we make `MustacheSequence` conform to `Sequence` then we don't need to make these function requirements of the protocol and can just implement them in `extension MustacheSequence {`. All types that conform to `MustacheSequence` conform to `Sequence` so you can always use the default implementations. Also they don't need to be public

Reply:
Hmm right. Thought that'd be too sneaky, but considering the comments I put there, it should be fine.